src/basic/device-nodes.h

```diff
@@ -22,5 +22,13 @@
 #include <stddef.h>
 #include <sys/types.h>
 
+#include "macro.h"
+#include "stdio-util.h"
+
 int encode_devnode_name(const char *str, char *str_enc, size_t len);
 int whitelisted_char_for_devnode(char c, const char *additional);
+
+#define SYS_BLOCK_PATH_MAX(suffix)                                      \
+        (strlen("/sys/dev/block/") + DECIMAL_STR_MAX(dev_t) + 1 + DECIMAL_STR_MAX(dev_t) + strlen(suffix))
```

Comment:
maybe use strlen_ptr(suffix) here? (to permit NULL suffix, as suggested above)

Reply:
OK.